src/mds/MDSRank.cc

```diff
@@ -901,9 +901,22 @@ Session *MDSRank::get_session(Message *m)
 {
   Session *session = static_cast<Session *>(m->get_connection()->get_priv());
   if (session) {
+    session->put(); // do not carry ref
     dout(20) << "get_session have " << session << " " << session->info.inst
 	     << " state " << session->get_state_name() << dendl;
-    session->put();  // not carry ref
+    // Check if we've imported an open session since
+    Session *imported_session = sessionmap.get_session(session->info.inst.name);
+    if (imported_session && imported_session != session) {
+      /* do not imported_session->put() because set_priv consumes ref */
+      dout(10) << __func__ << " replacing connection bootstrap session " << session << " with imported session " << imported_session << dendl;
+      imported_session->info.auth_name = session->info.auth_name;
+      //assert(session->info.auth_name == imported_session->info.auth_name);
+      assert(session->info.inst == imported_session->info.inst);
+      imported_session->connection = session->connection;
+      assert(session->get_nref() == 1);
```

Comment:
should be assert(session->get_nref() == 2);

Reply:
I think it's actually one here but it may be 2 with your suggested changes. I'll check...